vspace/src/main/java/edu/asu/diging/vspace/core/aspects/PreviewDataAspect.java

```diff
@@ -0,0 +1,65 @@
+package edu.asu.diging.vspace.core.aspects;
+
+import java.util.Arrays;
+import java.util.Map;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.annotation.After;
+import org.aspectj.lang.annotation.Around;
+import org.aspectj.lang.annotation.Aspect;
+import org.aspectj.lang.reflect.MethodSignature;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.ui.Model;
+import org.springframework.web.context.request.RequestContextHolder;
+import org.springframework.web.context.request.ServletRequestAttributes;
+import org.springframework.web.servlet.HandlerMapping;
+
+import edu.asu.diging.vspace.core.model.impl.Exhibition;
+import edu.asu.diging.vspace.core.services.IExhibitionManager;
+import edu.asu.diging.vspace.web.exhibit.view.ExhibitionConstants;
+
+/**
+ * 
+ * Represents Preview data aspect.
+ * @author prachikharge
+ *
+ */
+@Component
+@Aspect
+public class PreviewDataAspect {
+
+    @Autowired
+    private IExhibitionManager exhibitionManager;
+
+    /**
+     * Checks preview before and after joint point
```

Comment:
That's not really helpful. The javadoc should explain what this method does:
- checking if the requested url is for the preview (if it contains a preview id)
- checking if the preview id is correct (if not showing a 404)
- adding the preview id into the model and setting `isExhPreview` to true